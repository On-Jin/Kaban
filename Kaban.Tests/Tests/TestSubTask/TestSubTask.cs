using System.Text.Json;
using Kaban.Tests.Setup;
using Snapshooter;
using Snapshooter.Xunit;
using Xunit.Abstractions;

namespace Kaban.Tests.Tests.TestSubTask;

[Collection(nameof(WebAppCollectionFixture))]
public class TestSubTask(WebAppFactory factory, ITestOutputHelper testOutputHelper)
    : TestSubTaskBase(factory, testOutputHelper)
{
    [Fact]
    public async Task GraphQL_AddSubTask()
    {
        await ResetAndPopulateDb();

        {
            var response = await AddSubTask(HttpClientShadow);

            var jsonWrapped = await response.Content.ReadAsStringAsync();
            TestOutputHelper.WriteLine(jsonWrapped);

            JsonSerializer
                .Serialize(JsonSerializer.Deserialize<object>(jsonWrapped), TestHelper.JsonSerializerOptions)
                .MatchSnapshot();
        }
        PrettyJson(await GetQueryBoardsString()).MatchSnapshot(SnapshotNameExtension.Create("FullBoardQuery"));
    }
    
    [Fact]
    public async Task GraphQL_AddSubTasks()
    {
        await ResetAndPopulateDb();

        {
            var response = await AddSubTasks(HttpClientShadow);

            var jsonWrapped = await response.Content.ReadAsStringAsync();
            TestOutputHelper.WriteLine(jsonWrapped);

            JsonSerializer
                .Serialize(JsonSerializer.Deserialize<object>(jsonWrapped), TestHelper.JsonSerializerOptions)
                .MatchSnapshot();
        }
        PrettyJson(await GetQueryBoardsString()).MatchSnapshot(SnapshotNameExtension.Create("FullBoardQuery"));
    }
    
    [Fact]
    public async Task GraphQL_DeleteSubTask()
    {
        await ResetAndPopulateDb();

        {
            var response = await DeleteSubTask(HttpClientShadow);

            var jsonWrapped = await response.Content.ReadAsStringAsync();
            TestOutputHelper.WriteLine(jsonWrapped);

            JsonSerializer
                .Serialize(JsonSerializer.Deserialize<object>(jsonWrapped), TestHelper.JsonSerializerOptions)
                .MatchSnapshot();
        }
        PrettyJson(await GetQueryBoardsString()).MatchSnapshot(SnapshotNameExtension.Create("FullBoardQuery"));
    }
    
    [Fact]
    public async Task GraphQL_DeleteSubTasks()
    {
        await ResetAndPopulateDb();

        {
            var response = await DeleteSubTasks(HttpClientShadow);

            var jsonWrapped = await response.Content.ReadAsStringAsync();
            TestOutputHelper.WriteLine(jsonWrapped);

            JsonSerializer
                .Serialize(JsonSerializer.Deserialize<object>(jsonWrapped), TestHelper.JsonSerializerOptions)
                .MatchSnapshot();
        }
        PrettyJson(await GetQueryBoardsString()).MatchSnapshot(SnapshotNameExtension.Create("FullBoardQuery"));
    }
}