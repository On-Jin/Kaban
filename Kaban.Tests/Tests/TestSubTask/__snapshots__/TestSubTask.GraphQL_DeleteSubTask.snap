{
  "data": {
    "deleteSubTask": {
      "mainTask": {
        "description": "",
        "id": 1,
        "status": "Todo",
        "title": "Build UI for onboarding flow",
        "subTasks": [
          {
            "id": 2,
            "isCompleted": false,
            "title": "Sign in page"
          },
          {
            "id": 3,
            "isCompleted": false,
            "title": "Welcome page"
          }
        ]
      }
    }
  }
}
